app/navigation/navigation_modal.js

```diff
@@ -24,16 +24,20 @@ export default class NavigationModal extends PureComponent {
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.show === nextProps.show) {
+        const nextRoutesLength = nextProps.children.props.navigationState.routes.length;
+        const currentRoutesLength = this.state.children ? this.state.children.props.navigationState.routes.length : 0;
+        const routesDidChange = nextRoutesLength !== currentRoutesLength;
```

Comment:
I can't get it to work with just the equality check. Since the state is immutable ```routesDidChange``` will always be true since it will be different from what's stored in ```this.state.children```.